src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -449,6 +449,19 @@ def _get_extension_suppressions(mod_loaders):
                 if raw_cmd in self.command_group_table:
                     logger.debug("Found a match in the command group table for '%s'.", raw_cmd)
                     return self.command_table
+                if self.cli_ctx.data['completer_active']:
+                    # If the command is not complete in autocomplete mode, we should match shorter command.
+                    # For example, `account sho` should match `account`.
+                    logger.debug("Could not find a match in the command or command group table for '%s'", raw_cmd)
+                    raw_cmd = ' '.join(raw_cmd.split()[:-1])
```

Comment:
Please avoid reusing existing name. Instead, introduce a new name to make the code easier to read.